### PR TITLE
Provide the remote name to the TCPMetrics connected callback so clien…

### PR DESCRIPTION
**Pull request by vietj**
…t metrics can be reported using the host name

### CODE REVIEW

**Comment by tsegismont** [comment]
I haven't checked, do you know if the issue applies to DatagramSocketMetrics as well?

**Comment by vietj** [comment]
the is no connection notion in datagrams.

**Comment by vietj** [comment]
you mean void bytesRead(S socketMetric, SocketAddress remoteAddress, long numberOfBytes); ?

**Comment by vietj** [comment]
we can try to address these as well

**Comment by vietj** [comment]
WDYT ?

**Comment by tsegismont** [comment]
Right, no connection, but I was talking about the ability to get the host string provided by the user when he's sending UDP packets to a remote host

**Comment by vietj** [comment]
I can do these as well today I think.

**Comment by tsegismont** [comment]
I believe it also would be better to get the host string provided by the user when he's setting up a UDP listener

File: src/test/java/io/vertx/test/core/MetricsTest.java [hunk 1]
```diff
@@ -476,7 +478,16 @@ public void testMulti() {
   }
 
   @Test
-  public void testHttpConnect() {
+  public void testHttpConnect1() throws Exception {
+    testHttpConnect("localhost", socketMetric -> assertEquals("localhost", socketMetric.remoteName));
```
Comment (tsegismont) [inline src/test/java/io/vertx/test/core/MetricsTest.java]: So here the lambda is the check for the client metric, right?

File: src/test/java/io/vertx/test/core/MetricsTest.java [hunk 2]
```diff
@@ -493,15 +504,17 @@ public void testHttpConnect() {
         assertFalse(serverMetric.socket.connected.get());
         assertEquals(5, serverMetric.socket.bytesRead.get());
         assertEquals(5, serverMetric.socket.bytesWritten.get());
+        assertEquals(serverMetric.socket.remoteAddress.host(), serverMetric.socket.remoteName);
```
Comment (tsegismont) [inline src/test/java/io/vertx/test/core/MetricsTest.java]: I guess this is ok to test the server metric side, provided Netty does not resolve the address internally.

**Comment by tsegismont** [comment]
Looks good to me.

**Pull request merged**